wikimedia/wikipedia-ios
# Use `Configuration.Domain.localhost` for `SchemeHandler`'s local PCS configuration

**Phabricator ticket**: N/A

### Notes
Currently, `SchemeHandler` uses a hardcoded "localhost" to determine whether the resource scheme should be `http` or `https` on local PCS builds. This PR replaces that with the existing `Configuration.Domain.localhost` string.

Other than just replacing a magic string, this also has the added benefit of centralizing where a change needs to be made for development deployments of PCS. For example, if one switches from deploying PCS locally to anywhere else, they'd now only have to make a change to the `Configuration.Domain.localhost` for everything to work.

(This PR is based off of 6.6.1, but happy to change the merge base to master as it's not specifically critical for 6.6.1.)

### Test Steps (optional)
These are high commitment test steps for a low impact PR, so I really do not recommend or expect that anyone actually attempt anything other than maybe steps 1 and 2. Just documenting them:
1. Deploy `mobileapps` locally
2. Using the local PCS target, search for any article and confirm it loads as expected
3. Deploy `mobileapps` somewhere that isn't your local development machine
4. Update `Configuration.Domain.localhost`'s value to that new hostname
5. Using the local PCS target, search for any article and confirm it loads as expected
